app/grandchallenge/discussion_forums/models.py

```diff
@@ -176,44 +176,115 @@ class Meta:
             "created",
         ]
 
-    def __str__(self):
-        return self.subject
-
     @property
     def is_alone(self):
         return self.topic.posts.count() == 1
 
+    @property
+    def is_last_post(self):
+        return self.topic.last_post == self
+
     def save(self, *args, **kwargs):
+        adding = self._state.adding
+
         super().save(*args, **kwargs)
+
+        if adding:
+            self.assign_permissions()
+
         self.topic.last_post_on = self.created
         self.topic.save()
 
     def delete(self, *args, **kwargs):
+        update_last_post_on = False
+        topic = None
+
         if self.is_alone:
             self.topic.delete()
-        else:
-            super().delete(*args, **kwargs)
+        elif self.is_last_post:
+            update_last_post_on = True
+            topic = self.topic
+
+        super().delete(*args, **kwargs)
 
+        if update_last_post_on:
+            topic.last_post_on = topic.last_post.created
```

Comment:
Maybe it makes sense that a query is done here, or a separate task launched to update this? I imagine that in most cases `last_post` can be cached (or it would seem to a future developer an easy thing to cache).

Reply:
Good point, I will turn this into a query.